src/RA_httpthread.cpp

```diff
@@ -247,32 +247,55 @@ void RAWeb::SetUserAgentString()
 #ifndef NTSTATUS
     #define NTSTATUS long
 #endif
-    using fnRtlGetVersion = NTSTATUS(NTAPI*)(PRTL_OSVERSIONINFOEXW lpVersionInformation);
-    const auto& ntModule{ ::GetModuleHandleW(L"ntdll.dll") };
-    if (ntModule)
     {
-        auto RtlGetVersion = (fnRtlGetVersion)GetProcAddress(ntModule, "RtlGetVersion");
-        RTL_OSVERSIONINFOEXW osVersion = {};
+        using fnRtlGetVersion = NTSTATUS(NTAPI*)(PRTL_OSVERSIONINFOEXW lpVersionInformation);
+
+        const auto& ntModule{ ::GetModuleHandleW(L"ntdll.dll") };
+        if (!ntModule)
+            return;
+
+        RTL_OSVERSIONINFOEXW osVersion{};
+        const auto& RtlGetVersion{
+            reinterpret_cast<fnRtlGetVersion>(::GetProcAddress(ntModule, "RtlGetVersion"))
+        };
+        if (!RtlGetVersion)
+            return;
         RtlGetVersion(&osVersion);
-        if (osVersion.dwMajorVersion > 0)
+        // needs to be XP or higher, NT 5.1.2600 is XP AND XP w/SP3, SP1&2 is different, 5.0 is
+        // Windows 2000
+        if ((osVersion.dwMajorVersion < 5UL) &&
+            (osVersion.dwMinorVersion < 1UL) &&
+            (osVersion.dwBuildNumber  < 2600UL))
+            return;
         {
-            sUserAgent.append("WindowsNT ");
-            sUserAgent.append(std::to_string(osVersion.dwMajorVersion));
-            sUserAgent.append(".");
-            sUserAgent.append(std::to_string(osVersion.dwMinorVersion));
+            std::ostringstream oss;
+            oss << "WindowsNT " << osVersion.dwMajorVersion << '.' << osVersion.dwMinorVersion
+                << '.' << osVersion.dwBuildNumber;
+            sUserAgent.append(oss.str());
```

Comment:
Applied previous feedback but expanded on it. Build numbers can help us know what version of Windows someone is using more accurately.

Reply:
This was modeled after IE11 and Chrome user agent strings, neither of which include the build number.

Reply:
Why use `osstringstream` to create an intermediate string instead of just appending directly to the `sUserAgent` string as was previously done before?

Reply:
The build number will go away but I still think `ostringstream `should be used.

Basically what I've been repeating several times, there is more sources than this one
https://www.thecodingdelight.com/string-cplusplus/

Some Key Points if it's too long to read, I do research all the time. It's really important and feel like it's been quite understated on how important it really is.
- `Do not concatenate using `+ `operator. Use `+=` or `append()`.`
- Only use `stringstream `when you need to convert `other data types` `(E.g. int)` to `strings`
- Replace all instances of `+ `with `+=` where applicable.